packages/perfect-freehand/src/getStrokeOutlinePoints.ts

```diff
@@ -217,7 +209,14 @@ export function getStrokeOutlinePoints(
       draw a cap at the current point.
     */
 
-    if (nextDpr < 0) {
+    const nextVector = (i < points.length - 1? points[i + 1] : points[i]).vector
+    const nextDpr = (i < points.length - 1? dpr(vector, nextVector) : 1.0)
+    const prevDpr = dpr(vector, prevVector)
```

Comment:
Is prevDpr always equal to last iteration's nextDpr?

Reply:
Maybe! I'll check. Definitely a cleaner fix in here somewhere

Reply:
Actually they're a bit different: 
![image](https://user-images.githubusercontent.com/15892272/185960548-d865f642-2f6e-4f61-8ae0-1f8d0bce3915.png)

And the important ones:
![image](https://user-images.githubusercontent.com/15892272/185960796-f33b8be0-fb57-4ebf-8ae4-dac97b8768ee.png)

Reply:
Wait, yes they are!

![image](https://user-images.githubusercontent.com/15892272/185961152-0152e290-9ee6-45f9-888d-de5c3e744079.png)

Reply:
Except in really rare cases: 
![image](https://user-images.githubusercontent.com/15892272/185961398-1c5c50b4-0f40-4155-b3b3-f7c137aa4671.png)

Reply:
I believe it's when the sharp corner is the final point.

Reply:
Wait that's not quite right, disregard that

Reply:
Ok so yes, they are the same (apart from some float imprecision)
![image](https://user-images.githubusercontent.com/15892272/185964498-d0cb2ecd-3c9a-44a9-b887-da0987c7f3a1.png)

Except for very rare cases:
![image](https://user-images.githubusercontent.com/15892272/185964460-c681408b-2924-4932-8fa3-6b445743f0ec.png)

Reply:
Here's another example that's not from the final two iterations: 

![image](https://user-images.githubusercontent.com/15892272/185965616-82ae331a-0c82-4aab-9639-f0eb564a6811.png)